### PR TITLE
Implemented write_spmatrix_to_sparse_tensor

### PR DESCRIPTION
**Pull request by meownoid**
Implemented `write_numpy_to_sparse_tensor` which supports any `scipy.sparse` matrix (#27).

### CODE REVIEW

File: src/sagemaker/amazon/common.py [hunk 1]
```diff
@@ -89,6 +108,46 @@ def write_numpy_to_dense_tensor(file, array, labels=None):
         _write_recordio(file, record.SerializeToString())
 
 
+def write_numpy_to_sparse_tensor(file, array, labels=None):
```
Comment (owen-t) [inline src/sagemaker/amazon/common.py]: Should this be write_spmatrix_to_sparse_tensor ?

File: src/sagemaker/amazon/common.py [hunk 2]
```diff
@@ -89,6 +108,46 @@ def write_numpy_to_dense_tensor(file, array, labels=None):
...
+                             labels.shape, array.shape))
+        resolved_label_type = _resolve_type(labels.dtype)
+    resolved_type = _resolve_type(array.dtype)
+
+    csr_array = array.tocsr()
+    n_rows, n_cols = csr_array.shape
+
+    record = Record()
+    for row_idx in range(n_rows):
+        record.Clear()
```
Comment (owen-t) [inline src/sagemaker/amazon/common.py]: Is there a reason this isn't done at the end of the for-loop?

File: src/sagemaker/amazon/common.py [hunk 3]
```diff
@@ -89,6 +108,46 @@ def write_numpy_to_dense_tensor(file, array, labels=None):
         _write_recordio(file, record.SerializeToString())
 
 
+def write_numpy_to_sparse_tensor(file, array, labels=None):
+    """Writes a numpy array to a dense tensor"""
```
Comment (jesterhazy) [inline src/sagemaker/amazon/common.py]: spmatrix not numpy array

File: src/sagemaker/amazon/common.py (hunk 1 above)
Comment (jesterhazy) [reply src/sagemaker/amazon/common.py]: +1

File: src/sagemaker/amazon/common.py (hunk 2 above)
Comment (meownoid) [reply src/sagemaker/amazon/common.py]: Full analogy to the `write_numpy_to_dense_tensor` function.

https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/amazon/common.py#L84

File: src/sagemaker/amazon/common.py (hunk 1 above)
Comment (meownoid) [reply src/sagemaker/amazon/common.py]: Ok, fixed.

**Comment by owen-t** [comment]
Thanks again for your submission

**Comment by andremoeller** [comment]
@meownoid Merged in, thanks!

**Pull request merged**